content/v2.0/router/advanced-usage.md

```diff
@@ -0,0 +1,624 @@
+    <td>:create</td>
+    <td>:books</td>
+  </tr>
+  <tr>
+    <td>GET</td>
+    <td>/books/:id/edit</td>
+    <td>Books::Edit</td>
+    <td>:edit</td>
+    <td>:edit_book</td>
+  </tr>
+  <tr>
+    <td>PATCH</td>
+    <td>/books/:id</td>
+    <td>Books::Update</td>
+    <td>:update</td>
+    <td>:book</td>
+  </tr>
+  <tr>
+    <td>DELETE</td>
+    <td>/books/:id</td>
+    <td>Books::Destroy</td>
+    <td>:destroy</td>
+    <td>:book</td>
+  </tr>
+</table>
+
+### Remove Routes
+
+In case we don"t need all the default routes we can use `:only` and pass one or more action names.
+We can also black list routes with `:except`.
+
+```ruby
+# /config/routes.rb
+
+Hanami.applicaton.routes do
+  resources :books, only: [:new, :create, :show]
+
+  # equivalent to
+
+  resources :books, except: [:index, :edit, :update, :destroy]
+end
+```
+
+### Add Routes
+
+Alongside with default routes we can specify extra routes for single (`member`) or multiple (`collection`) resources.
+
+```ruby
+# /config/routes.rb
+
+Hanami.applicaton.routes do
+  resources :books do
+    member do
+      # Generates /books/1/toggle, maps to Books::Toggle, named :toggle_book
+      get "toggle"
+    end
+
+    collection do
+      # Generates /books/search, maps to Books::Search, named :search_books
+      get "search"
+    end
+  end
+end
+```
+
+### Configure Controller
+
+Imagine we have a controller named `manuscripts`, where we have actions like `Manuscripts::Index`, but we still want to expose those resources as `/books`.
+Using the `:controller` option will save our day.
+
+```ruby
+# /config/routes.rb
+
+Hanami.applicaton.routes do
+  resources :books, controller: "manuscripts"
+
+  # GET /books/1 will route to Manuscripts::Show, etc.
+end
+```
+
+## RESTful Resource
+
+You can also use a `resource` helper, to generate routes for singular resource.
+
+```ruby
+# /config/routes.rb
+
+Hanami.application.routes do
+  resource :account
+end
+```
+
+It generates:
+
+<table class="table table-bordered table-striped">
+  <tr>
+    <th>Verb</th>
+    <th>Path</th>
+    <th>Action</th>
+    <th>Name</th>
+    <th>Named Route</th>
+  </tr>
+  <tr>
+    <td>GET</td>
+    <td>/account</td>
+    <td>Account::Show</td>
+    <td>:show</td>
+    <td>:account</td>
+  </tr>
+  <tr>
+    <td>GET</td>
+    <td>/account/new</td>
+    <td>Account::New</td>
+    <td>:new</td>
+    <td>:new_account</td>
+  </tr>
+  <tr>
+    <td>POST</td>
+    <td>/account</td>
+    <td>Account::Create</td>
+    <td>:create</td>
+    <td>:account</td>
+  </tr>
+  <tr>
+    <td>GET</td>
+    <td>/account/edit</td>
+    <td>Account::Edit</td>
+    <td>:edit</td>
+    <td>:edit_account</td>
+  </tr>
+  <tr>
+    <td>PATCH</td>
+    <td>/account</td>
+    <td>Account::Update</td>
+    <td>:update</td>
+    <td>:account</td>
+  </tr>
+  <tr>
+    <td>DELETE</td>
+    <td>/account</td>
+    <td>Account::Destroy</td>
+    <td>:destroy</td>
+    <td>:account</td>
+  </tr>
+</table>
+
+### Remove Routes
```

Comment:
There's a lot of duplicated content in this file, from this point onward. Think a couple extra pastes might have gotten saved by accident :)